### ISSUE TITLE
react table breaks after updating next from 12.0.7 to 12.0.8

### ISSUE THREAD

**Issue by HT-Moh** [vercel/next.js#33500]
### Run `next info` (available from version 12.0.8 and up)

The bug is caused when next is upgraded from 12.0.7 to 12.0.8

### What version of Next.js are you using?

12.0.8

### What version of Node.js are you using?

12.0.0

### What browser are you using?

chronme

### What operating system are you using?

Not important 

### How are you deploying your application?

Not important 

### Describe the Bug

I am not sure if this should be report here or [react table](https://github.com/tannerlinsley/react-table/issues/3650). this why I am posting it in the two repos

The issue is simply I updated next from `next@12.0.7` to `next@12.0.8` and the react table stop working and does not display the data anymore, when I rollback to next@12.0.7 it works.

![issue-react-table](https://user-images.githubusercontent.com/34759569/150376021-c3c0e70e-3620-4215-9054-c99df0b2b28d.png)

As you see `td` and `tr` html are displayed but not the cell content,  I even removed the render function and just hard code the value, same result- 
```javascript
    <tbody {...getTableBodyProps()}>
          {page.map((row, index) => {
            prepareRow(row)
            const { key, ...restRowProperties } = row.getRowProps()
            return (
              // eslint-disable-next-line react/jsx-key
              <tr key={key} {...restRowProperties}>
                {row.cells.map((cell: Cell<PostFragment>) => {
                  const { key, ...restCellProperties} = cell.getCellProps()
                  return (
                    <td {...restCellProperties} key={key}>
                         {/* {cell.render('Cell')}{' '} */}
                        test
                    </td>
                  )
                })}
              </tr>
            )
          })}
        </tbody>
```
I am not sure really what is causing this issue, but for sure 12.0.8

### Expected Behavior

It should just work as with 12.0.7

### To Reproduce

Use an example for [react table](https://react-table.tanstack.com/docs/overview) with  "next": "^12.0.8",

**Comment by bradwoods** [comment]
hey mate, does this only happen when running in dev mode? If you build & run `npm run next`, do you still get the error?

**Comment by balazsorban44** [comment]
Hi @HT-Moh could you provide a small reproduction? :pray: Could you also try out `canary` as we recently shipped some fixes that might be relevant.

**Comment by balazsorban44** [comment]
I've tried using this basic official example, and I don't have an issue:

https://github.com/tannerlinsley/react-table/blob/master/examples/basic/src/App.js

**Comment by albertbronsky** [comment]
Can confirm this issue. 

`next@12.0.8` just does not work with `react-table` with both `dev` and `build`. 

Only rollback to `next@12.0.7` helps.

**Comment by JSin** [comment]
Refer to https://github.com/vercel/next.js/issues/33414. The issue should has been fixed since `v12.0.9-canary.2`.

**Comment by balazsorban44** [comment]
Thank you @JSin! I knew this looked familiar! 😅 closing as duplicate of #33414
